File: Stack/StackList.java

class Node {
    int data;
    Node next;
    Node() {
        data = 0;
        next = null;
    }

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class Stack {
    private Node top;

    public Stack() {
        this.top = null;
    }

    public void push(int data) {
        Node newNode = new Node(data);
        if(this.top == null) {
            this.top = newNode;
        }else{
            newNode.next = this.top;
            this.top = newNode;
        }
    }

    public int pop() {
        if(this.top == null) {
            return -1;
        }else{
            Node ptr = this.top;
            int data = ptr.data;
            this.top = ptr.next;
            ptr = null;
            return data;
        }
    }
}


public class StackList {
    public static void main(String[] args) {
        Stack stack = new Stack();
        stack.push(1);
        stack.push(2);
        stack.push(3);
        System.out.print(stack.pop()+" ");
        System.out.print(stack.pop()+" ");
        System.out.print(stack.pop()+" ");
        System.out.print(stack.pop()+" ");
    }
}

File: Stack/StackList.py
"""
Stack using linked list
"""

class Node : 

    def __init__(self, data) : 
        self.data = data
        self.next_node = None


class Stack : 
    
    def __init__(self) : 
        self.top = None
        

    
    def push(self, data) : 
        newNode = Node(data)

        if self.top is None : 
            self.top = newNode
        else :
            newNode.next_node = self.top
            self.top = newNode
        

    def pop(self) -> int : 
        if self.top is None : 
            return "Stack is empty"
        
        ptr = self.top 
        temp = ptr.data
        self.top = ptr.next_node
        del ptr
        return temp



if __name__ == '__main__' : 
    stack = Stack()

    stack.push(1)
    stack.push(2)
    stack.push(3)

    print(stack.pop())
    print(stack.pop())
    print(stack.pop())
    print(stack.pop())
        

